core/src/test/scala-3/org/scalacheck/ops/ArbitraryAsGenSpec.scala

```diff
@@ -0,0 +1,23 @@
+package org.scalacheck.ops
+
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.freespec.AnyFreeSpec
+
+class ArbitraryAsGenSpec extends AnyFreeSpec {
+
+  private val it = classOf[ArbitraryAsGen].getSimpleName
+
+  s"$it should implicitly convert a Gen to Arbitrary" in {
+    def f(arb: Arbitrary[Char]): Unit = ()
+    f(Gen.numChar)
+  }
+
+  private def implicitGen(implicit gen: Gen[Char]): Unit = ()
+
+  s"$it should convert an implicit Arbitrary to an implicit Gen" in {
+    /*
+     * Note this compiles in Scala 3, not Scala 2
+     */
+    assertCompiles("implicitGen")
```

Comment:
I'm not sure I like this "feature" anymore... but I don't want to complicate this PR, just noting that if it is easier to remove the "feature" for Scala 3, I wouldn't mind it.